ajax/timeline.php

```diff
@@ -56,19 +56,19 @@
         $new_state = ($task->fields['state'] == Planning::DONE)
                         ? Planning::TODO
                         : Planning::DONE;
-        $new_label = Planning::getState($new_state);
-        echo json_encode([
-            'state'  => $new_state,
-            'label'  => $new_label
-        ]);
-
         $foreignKey = $parent->getForeignKeyField();
         $task->update([
             'id'        => intval($_POST['tasks_id']),
             $foreignKey => intval($_POST[$foreignKey]),
             'state'     => $new_state,
             'users_id_editor' => Session::getLoginUserID()
         ]);
+        $task->getFromDB(intval($_POST['tasks_id']));
```

Comment:
Object will be automatically reloaded if an update is done.

Unless I miss something, the only case that may cause an issue if that item is not reloaded if update query fails (bad SQL syntax, connection error, ...). If a fix have to be done on this, it should be a global fix (an probably in another PR).

```suggestion
```